src/Catty/FormulaEditor/ViewController/FormulaEditorSectionViewController.swift

```diff
@@ -0,0 +1,405 @@
+        self.numberOfSections = numberOfRowsInSection.count
+        self.titlesOfSections = [FunctionSubsection.maths.title, FunctionSubsection.texts.title, FunctionSubsection.lists.title]
+    }
+
+    private func initLogicItems() {
+        self.items.removeAll()
+
+        if let object = spriteObject, let manager = formulaManager {
+            self.items = manager.formulaEditorItemsForLogicSection(spriteObject: object)
+        }
+
+        self.numberOfRowsInSection = self.groupSubsectionWiseAndGetSize(items.first?.sections.first?.subsection() as? LogicSubsection, items: &items)
+        self.numberOfSections = numberOfRowsInSection.count
+        self.titlesOfSections = [LogicSubsection.logical.title, LogicSubsection.comparison.title]
+
+    }
+
+    private func initObjectItems() {
+        self.items.removeAll()
+
+        if let object = spriteObject, let manager = formulaManager {
+            self.items = manager.formulaEditorItemsForObjectSection(spriteObject: object)
+        }
+
+        self.numberOfRowsInSection = self.groupSubsectionWiseAndGetSize(items.first?.sections.first?.subsection() as? ObjectSubsection, items: &items)
+        self.numberOfSections = numberOfRowsInSection.count
+        self.titlesOfSections = [ObjectSubsection.general.title, ObjectSubsection.motion.title]
+
+    }
+
+    private func initSensorItems() {
+        self.items.removeAll()
+
+        if let object = spriteObject, let manager = formulaManager {
+            self.items = manager.formulaEditorItemsForDeviceSection(spriteObject: object)
+        }
+
+        self.numberOfRowsInSection = self.groupSubsectionWiseAndGetSize(items.first?.sections.first?.subsection() as? SensorSubsection, items: &items)
+        self.numberOfSections = numberOfRowsInSection.count
+        self.titlesOfSections = [SensorSubsection.device.title,
+                                 SensorSubsection.touch.title,
+                                 SensorSubsection.visual.title,
+                                 SensorSubsection.dateAndTime.title,
+                                 SensorSubsection.arduino.title,
+                                 SensorSubsection.phiro.title]
+    }
+
+    private func groupSubsectionWiseAndGetSize<SubsectionType: FormulaEditorSubsection & Hashable & CaseIterable>(_ subsectionType: SubsectionType?, items: inout [FormulaEditorItem]) -> [Int] {
+
+        guard let _ = subsectionType else {
+            return [Int]()
+        }
+
+        var sizes = [Int]()
+        let dict = Dictionary(grouping: items, by: { $0.sections[0].subsection() as! SubsectionType })
+        items.removeAll()
+
+        for subsection in SubsectionType.allCases {
+            if let groupedItems = dict[subsection] {
+                items.append(contentsOf: groupedItems)
+                sizes.append(groupedItems.count)
+            }
+        }
+
+        return sizes
+    }
+
+    private func initDataItems() {
```

Comment:
There still seems to be a lot of code which is solely for the data section. So I was wondering: would it be more readable if we create a new VC class for this section (could be derived from the FESectionVC). Maybe this could also be an idea for all other sections (e.g., in order to get rid of the RawValue enum workaround). What do you think?

Reply:
Thanks! This sounds awesome! 
PS: I also did something like this for the keyboard buttons that you'll see in CATTY-517!

Reply:
However, I think this should only be done for the data section, as behaviour of all other sections is identical. But in that case we will still need enums. What's your suggestion on that?

Reply:
Perfect, the data section is also the most important one for me. The others do differ in the initialization which is also not negligible and would be enough for me to create a new class, but that must not be. Please only go for the data section at the moment. Thanks!

Reply:
I agree, separate classes for each section would be a better approach than enums. Please review the new `FormulaEditorDataSectionViewController`, if it looks good, I shall implement similar classes for all other sections too.

Reply:
Please see my comments directly at the files. Both approaches are fine for me -- if you keep the enums, could you please get rid of the `.none` enum?

Reply:
I had kept `.none` case inside this enum because it was serving for the "gap" by the inheriting classes (such as, FormulaEditorDataSectionViewController). Additionally, I can't have that property as an optional, because optional enums are not represented in Objective-C. So I have added back the `.data` case again, and updated the initialisers. However, I am not sure whether this is the best solution. What do you think?

Reply:
Thanks for the explanation! I agree that it is not perfect, but also that it is not easily solvable in a better way with the current architecture. For me this approach is fine.